addons/point_of_sale/models/pos_session.py

```diff
@@ -408,6 +412,9 @@ def _create_account_move(self):
                         stock_expense[exp_key] = self._update_amounts(stock_expense[exp_key], {'amount': amount}, move.picking_id.date)
                         stock_output[out_key] = self._update_amounts(stock_output[out_key], {'amount': amount}, move.picking_id.date)
 
+            # Setting current partner's customer_rank to 1 if necessary
+            order.partner_id._increase_rank('customer_rank')
```

Comment:
```suggestion
if order.partner_id:
      order.partner_id._increase_rank('customer_rank')
```

The customer_rank is not set to 1, but incresed by 1, maybe change the message.

Reply:
i place the check higher 
https://github.com/odoo/odoo/blob/24176e8c06c61df378614deea565ad28daca7bfa/addons/account/models/partner.py#L510